core/controllers/admin_test.py

```diff
@@ -128,45 +128,49 @@ def test_change_rights(self):
         self.signup(
             BOTH_MODERATOR_AND_ADMIN_EMAIL, BOTH_MODERATOR_AND_ADMIN_USERNAME)
 
-        # Navigate to any page. The role is not set.
-        self.testapp.get('/').mustcontain(no=['/moderator', '/admin'])
-
-        # Log in as a superadmin. This gives access to /admin.
-        self.login('superadmin@example.com', is_super_admin=True)
-        self.testapp.get('/').mustcontain('/admin', no=['/moderator'])
-
-        # Add a moderator, an admin, and a person with both roles, then log
-        # out.
-        response = self.testapp.get('/admin')
-        csrf_token = self.get_csrf_token_from_response(response)
-        self.post_json('/adminhandler', {
-            'action': 'save_config_properties',
-            'new_config_property_values': {
-                config_domain.ADMIN_USERNAMES.name: [
-                    self.ADMIN_USERNAME,
-                    BOTH_MODERATOR_AND_ADMIN_USERNAME],
-                config_domain.MODERATOR_USERNAMES.name: [
-                    self.MODERATOR_USERNAME,
-                    BOTH_MODERATOR_AND_ADMIN_USERNAME],
-            }
-        }, csrf_token)
-        self.logout()
-
-        # Log in as a moderator.
-        self.login(self.MODERATOR_EMAIL)
-        self.testapp.get(feconf.GALLERY_URL).mustcontain(
-            '/moderator', no=['/admin'])
-        self.logout()
-
-        # Log in as an admin.
-        self.login(self.ADMIN_EMAIL)
-        self.testapp.get(feconf.GALLERY_URL).mustcontain(
-            '/moderator', no=['/admin'])
-        self.logout()
-
-        # Log in as a both-moderator-and-admin. Only '(Admin)' is shown in the
-        # navbar.
-        self.login(BOTH_MODERATOR_AND_ADMIN_EMAIL)
-        self.testapp.get(feconf.GALLERY_URL).mustcontain(
-            '/moderator', no=['/admin'])
-        self.logout()
+        if self.testapp.get('/').status_int != 302:
```

Comment:
@seanlip I had a query:
In the current codebase, '/' does not have a separate page associated with it (it is directly pointed to gallery page now).

But now since checking for user login status has to be made for '/', so a redirect will have to be made, either to '/gallery' (if not logged in) or to '/my_explorations' (if logged in).  But it will always be a redirect.
Is this approach correct ?